### PR TITLE
B361 - Non premium users should not be able to host a queue.

### PR DESCRIPTION
**Pull request by creiser2**

### CODE REVIEW

**Comment by creiser2** [comment]
With this PR, free users app remote must be disconnected immediately, otherwise will break as app remote connection think its successful, but is not.

File: CranberryQueue/Map/MapViewController.swift [hunk 1]
```diff
@@ -274,7 +274,22 @@ class MapViewController: UIViewController, UITextFieldDelegate, MapControllerDel
...
             if connected {
-                openCreateQueueModal()
+                if isPremium == nil {
+                    checkUserProduct() { premium in
+                        self.isPremium = premium
+                        if premium == false {
+                            DispatchQueue.main.async {
+                                self.connectionStatusInvoker = .none
+                                self.showPremiumAlert()
+                                (UIApplication.shared.delegate as? AppDelegate)?.appRemote.disconnect()
```
Comment (rolflocher) [inline CranberryQueue/Map/MapViewController.swift]: shouldnt this code be in updateSessionStatus, before the non-premium user attempts to connect their app remote?

right now all of this code is running after their app remote has already connected

File: CranberryQueue/Map/MapViewController.swift [hunk 2]
```diff
@@ -274,7 +274,22 @@ class MapViewController: UIViewController, UITextFieldDelegate, MapControllerDel
...
+                            }
+                        } else {
+                            self.openCreateQueueModal()
+                        }
+                    }
+                } else {
+                    openCreateQueueModal()
```
Comment (rolflocher) [inline CranberryQueue/Map/MapViewController.swift]: so this line fires if they are connected but not premium? consider using early returns to flatten this if-else tree

File: CranberryQueue/Map/MapViewController.swift [hunk 3]
```diff
@@ -306,14 +321,58 @@ class MapViewController: UIViewController, UITextFieldDelegate, MapControllerDel
...
-        createQueueForm.isHidden = false
-        UIView.animate(withDuration: 0.3) {
-            self.createQueueForm.alpha = 1
+        if isPremium == true {
+            DispatchQueue.main.async {
+                self.createQueueForm.isHidden = false
+                UIView.animate(withDuration: 0.3) {
+                    self.createQueueForm.alpha = 1
+                }
+                self.createQueueForm.queueNameTextField.becomeFirstResponder()
```
Comment (rolflocher) [inline CranberryQueue/Map/MapViewController.swift]: might want to reorder these three calls

Comment (rolflocher) [reply CranberryQueue/Map/MapViewController.swift]: inside dispatchQueue.main.async {

File: CranberryQueue/Map/MapViewController.swift [hunk 4]
```diff
@@ -244,6 +244,13 @@ class MapViewController: UIViewController, UITextFieldDelegate, MapControllerDel
     // Called when "+" icon tapped
     @objc func createQueueTapped() {
         recenterMapTapped()
+        
+        //show alert and stop action
+        if isPremium == false {
+            self.showPremiumAlert()
```
Comment (rolflocher) [inline CranberryQueue/Map/MapViewController.swift]: self. !!!

**Pull request merged**